### PR TITLE
Remove the warning when the runtime is Opal/JavaScript

### PR DESCRIPTION
**Pull request by Mogztter**
When registering a custom converter with Asciidoctor.js the following warning message is printed:

> asciidoctor: WARNING: gem 'concurrent-ruby' is not installed. This gem is recommended when registering custom converters.

This warning message does not make sense in Asciidoctor.js and shouldn't be printed to avoid confusion.

Note: I've added a test case (on my local workspace) but I don't know how to simulate the absence of `concurrent-ruby` and/or to fake an `opal` Ruby engine...

### CODE REVIEW

File: lib/asciidoctor/converter/factory.rb [hunk 1]
```diff
@@ -53,7 +53,9 @@ def default initialize_singleton = true
             new ::Concurrent::Hash.new
```
Comment (mojavelinux) [inline lib/asciidoctor/converter/factory.rb]: But isn't Asciidoctor.js providing Concurrent::Hash so that this LoadError isn't hit in the first place?

**Comment by Mogztter** [comment]
I've updated my pull request.
I think it's better to not execute code that does not make sense in an Opal environment. With this special condition, you will be able to remove the dynamic require (when Ruby 1.9 support will be dropped).

**Comment by mojavelinux** [comment]
> you will be able to remove the dynamic require (when Ruby 1.9 support will be dropped).

FYI, master has already dropped support for Ruby < 2.3.

**Comment by Mogztter** [comment]
@mojavelinux But why do we still have the following line:

```rb
require ::RUBY_MIN_VERSION_1_9 ? 'concurrent/hash' : 'asciidoctor/core_ext/1.8.7/concurrent/hash'
```

**Comment by mojavelinux** [comment]
Oh yeah, I just haven't merged it yet. Antora 2.0 release was holding me up. Asciidoctor 2.0 release is my next focus...call it my holiday project.

**Comment by mojavelinux** [comment]
Yep, I agree. Thanks!

**Pull request merged**